File: 0801-0900/0848-Shifting Letters/0848-Shifting Letters.cpp
class Solution {
public:
    string shiftingLetters(string S, vector<int>& shifts) {
        int n = S.size(), sum = 0;
        for (int i = n - 1; i >= 0; --i) {
            sum = (sum + shifts[i]) % 26;
            S[i] = 'a' + (S[i] - 'a' + sum) % 26;
        }
        return S;
    }
};

File: 0801-0900/0848-Shifting Letters/0848-Shifting Letters.py
class Solution:
    def shiftingLetters(self, S: str, shifts: List[int]) -> str:
        result = []
        s = 0
        for i in range(len(S) - 1, -1, -1):
            s = (s + shifts[i]) % 26
            result.append(chr(ord('a') + (ord(S[i]) - ord('a') + s) % 26))
        return ''.join(result[::-1])
